manta-util/src/num.rs

```diff
@@ -67,3 +67,58 @@ pub trait CheckedSub<Rhs = Self> {
     /// Checked integer subtraction. Computes `self - rhs`, returning `None` if overflow occurred.
     fn checked_sub(self, rhs: Rhs) -> Option<Self::Output>;
 }
+
+/// Checked Increment
+pub trait CheckedIncrement {
+    /// Increments `self` returning `None` if it would overflow.
+    fn checked_increment(&mut self) -> Option<&mut Self>;
+}
+
+/// Checked Decrement
+pub trait CheckedDecrement {
+    /// Decrements `self` returning `None` if it would underflow.
```

Comment:
In `CheckedSub` above we call this "overflow", so we should make the two consistent. [Wikipedia](https://en.wikipedia.org/wiki/Arithmetic_underflow) suggests "overflow" is the correct term here 

```suggestion
    /// Decrements `self` returning `None` if it would overflow.
```